webapp/src/components/viewHeader/viewHeader.tsx

```diff
@@ -37,19 +37,23 @@ type Props = {
 
 const ViewHeader = React.memo((props: Props) => {
     const [showFilter, setShowFilter] = useState(false)
-
     const {boardTree, showView} = props
     const {board, activeView} = boardTree
-
     const withGroupBy = activeView.viewType === 'board' || activeView.viewType === 'table'
-
     const [viewTitle, setViewTitle] = useState(activeView.title)
 
     useEffect(() => {
         setViewTitle(activeView.title)
     }, [activeView.title])
 
     const hasFilter = activeView.filter && activeView.filter.filters?.length > 0
+    const onCloseFilter = () => {
+        document.body.addEventListener('click', (event: Event) => {
```

Comment:
This is adding an even listerner on every filter close and is not removing it, also the check of the target as an string looks fragile.

I think there must be a better way to do this. What is the thinking behind this block of code?

Reply:
the problem here is that when we click Filter Button it triggers 2 events onClick and onClose (because filter button is outside the dropdown menu) and that's why the dropdown remains open.
I agree about the use of string